src/Commands.hs

```diff
@@ -808,8 +808,14 @@ commandNot ctx [x] =
 commandReadFile :: CommandCallback
 commandReadFile ctx [filename] =
   case filename of
-    XObj (Str fname) _ _ -> do
-         exceptional <- liftIO ((try $ slurp fname) :: (IO (Either IOException String)))
+    XObj (Str fname) i _ -> do
+         let relativeTo = case i of
```

Comment:
This is a copypaste from the [load](https://github.com/VPashkov/Carp/blob/7e440cae8ce1a40322fdd0e7db20b08c299ef429/src/Eval.hs#L612) command.
I'm not sure where to put it to keep it DRY. Or maybe it's OK to leave it here?

Reply:
You could move this code out into a helper function, for instance:

```haskell
getRelativeOf i =
  case i of
    Just ii ->
      case infoFile ii of
        "REPL" -> "."
        file -> takeDirectory file
    Nothing -> "."
```

And then call that there. I don’t mind the duplication too much, though!